api/common/common_types.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"github.com/openshift/assisted-service/models"
+)
+
+// +genclient
+
+// +kubebuilder:object:generate=true
+type ValidationResult struct {
+	ID      models.HostValidationID `json:"id"`
```

Comment:
Why not use string instead of models.HostValidationID ?
More than that, you will use this type in AgentClusterInstall , so make no sense to call that way.